src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -89,22 +114,30 @@ public SoftwareSpi(int clk, int miso, int mosi, int cs, SpiConnectionSettings se
             else
             {
                 _bitTransfer = new ScopeData(
+                    enter: () => { },
```

Comment:
why didn't this need null check before (if it did have it please remove the check)

Reply:
nvm, there was a null check